client/src/Features/MicrosoftLearn/MicrosoftLearnFilterTags.tsx

```diff
@@ -0,0 +1,79 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License.
+ *--------------------------------------------------------------------------------------------*/
+
+import React from 'react';
+import { useStore } from '../../Stores/Core';
+import { useObserver } from 'mobx-react-lite';
+import { FilterType } from '../../Models/Learn/FilterType.model';
+import { FilterTag, getDisplayFilterTags } from './MicrosoftLearnFilterUtils';
+import { DefaultButton, styled } from '@fluentui/react';
+import { IStylesOnly, IThemeOnlyProps, SimpleComponentStyles } from '../../Core/Utils/FluentUI/typings.fluent-ui';
+import { themedClassNames } from '../../Core/Utils/FluentUI';
+
+type FilterTagStyles = SimpleComponentStyles<'root' | 'tags'>;
+
+const FilterTagsInner = ({ styles }: IStylesOnly<FilterTagStyles>): JSX.Element | null => {
+  const { filterStore, catalog } = useStore('microsoftLearnStore');
+  const classes = themedClassNames(styles);
+
+  return useObserver(() => {
+    let tagMap: FilterTag[] = getDisplayFilterTags(
+      filterStore.displayFilter,
+      filterStore.selectedFilter,
+      filterStore.productMap,
+      catalog
+    );
+    return (
+      <div className={classes.root}>
+        {tagMap.map(tag => (
+          <DefaultButton
+            className={classes.tags}
+            iconProps={{ iconName: 'StatusCircleErrorX' }}
+            text={tag.name}
+            onClick={() => {
+              if (tag.type === FilterType.products) {
+                let subItems: string[] = [...catalog?.products.values()]
```

Comment:
I thought we could keep the `filterStore::removeFilter` common for all for filter types, without any separate checks for `products`. Although we can consider moving this in future?

Reply:
The reason I was looking at it that way is this seems to be a logic thing and remove can happen not just via this `chicklet` but via some other action as well. `FilterStore` at this moment is responsible for making sure right things get selected, added and displayed (via the Utils). Only, remove is the one that is now not completely responsible for, hence, that was feeling a bit out-of-place. We can consider taking this up in future for sure, just wondering if there's anything that seems like a friction in refactor (something we should be careful of when doing, which is why we plan to take it separately)?

Reply:
I don't think there's anything, in particular, we'll have to take care of while refactoring; I can do it in this iteration itself. I just had one small concern though- since it is a part of the logic, would we have to get it tested again, or a check from my end that nothing breaks up should do?

Reply:
I think we've refactored it fine for now. Moving to `FilterStore` could be done separately.